### ISSUE TITLE
Linting fails to complete when --project option is passed to tslint

### ISSUE THREAD

**Issue by andriy-kudrya-eleks** [SublimeLinter/SublimeLinter-tslint#22]
Several linting rules require --project option to be specified but when I set one linting stops functioning. ts-loader with similar config works fine.
Here is config from *.sublime-project*
```
"tslint": {
    "args": [
        "--project", "${project}/tsconfig.json",
        "--config", "${project}/config/tslint.json"
    ]
}
```
And debug output from sublime console:
```
SublimeLinter: tslint: index.ts ['D:\\src\\node_modules\\.bin\\tslint.cmd', '@', '--project', 'D:/src/tsconfig.json', '--config', 'D:/src/config/tslint.json'] 
SublimeLinter: tslint output:

Invalid source file: c:\users\user\appdata\local\temp\SublimeLinter3-user\index.ts. Ensure that the files supplied to lint have a .ts, .tsx, .d.ts, .js or .jsx extension. 
SublimeLinter: No match for <_sre.SRE_Pattern object at 0x000001BC10665A50> 
SublimeLinter: No match for <_sre.SRE_Pattern object at 0x000001BC10665A50> 
SublimeLinter: No match for <_sre.SRE_Pattern object at 0x000001BC10665A50> 

```
It happens on Windows 10. Without --project options linting works but uses some unspecified rules in tslint.json

**Comment by blieber** [comment]
Repros on mac as well

Invalid source file: ../../../../../../../../../../../../var/folders/3s/bpn0crpd3k11px62sm7s204rg5ftm1/T/SublimeLinter3-{user}/Filename.tsx. Ensure that the files supplied to lint have a .ts, .tsx, .d.ts, .js or .jsx extension. 
SublimeLinter: No match for <_sre.SRE_Pattern object at 0x10796aa50> 
SublimeLinter: No match for <_sre.SRE_Pattern object at 0x10796aa50> 
SublimeLinter: No match for <_sre.SRE_Pattern object at 0x10796aa50> 
SublimeLinter: tslint: Filename.tsx ['/Users/{user}/{path}/node_modules/.bin/tslint', '@', '--config', '/Users/{user}/{path}/tslint.json', '--project', '/Users/{user}/{path}/tsconfig.json'] 
SublimeLinter: tslint output:

**Comment by braver** [comment]
Can you guys try again with recent versions of SublimeLinter and this plugin? Everything has changed.

**Comment by andriy-kudrya-eleks** [comment]
Hi @braver . Sorry for the delay. Plugin seems to work properly now.
As a side note, there is no error in console when I supply wrong config path.
Also, there is an inconvenient thing - linting seems to occur only after a file has been saved.

**Comment by braver** [comment]
The config path thing is probably worth a new issue.

The on-save only thing is by design, it won’t be able to follow imports otherwise.
